### PR TITLE
ddl: add region and split compatibility for temporary table

### PR DESCRIPTION
**Pull request by Howie59**
### What problem does this PR solve?

Issue Number: see https://github.com/pingcap/tidb/pull/24168

### What is changed and how it works?

Proposal: [temporary table](https://github.com/pingcap/tidb/issues/24169) 

What's Changed:  ddl: add region and split compatibility for temporary table

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -3069,6 +3077,9 @@ func (b *PlanBuilder) buildIndexAdvise(node *ast.IndexAdviseStmt) Plan {
 }
 
 func (b *PlanBuilder) buildSplitRegion(node *ast.SplitRegionStmt) (Plan, error) {
+	if node.SplitSyntaxOpt != nil && node.Table.TableInfo.TempTableType != model.TempTableNone {
```
Comment (djshow832) [inline planner/core/planbuilder.go:3080]:
```suggestion
	if node.Table.TableInfo.TempTableType != model.TempTableNone {
```

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1925,6 +1925,9 @@ func (d *ddl) CreateTableWithInfo(
 // preSplitAndScatter performs pre-split and scatter of the table's regions.
 // If `pi` is not nil, will only split region for `pi`, this is used when add partition.
 func (d *ddl) preSplitAndScatter(ctx sessionctx.Context, tbInfo *model.TableInfo, pi *model.PartitionInfo) {
+	if tbInfo.TempTableType != model.TempTableNone {
+		return
```
Comment (djshow832) [inline ddl/ddl_api.go:1929]: It's better to report an error.
And thus, you can test it by checking an error in `create table t(id int ) pre_split_regions=2;`.

**Comment by Howie59** [comment]
PTAL @djshow832

File: planner/core/preprocess.go [hunk 3]
```diff
@@ -585,6 +585,21 @@ func (p *preprocessor) checkAdminCheckTableGrammar(stmt *ast.AdminStmt) {
...
+		if stmt.ReferTable.Schema.String() != "" {
+			schema = stmt.ReferTable.Schema
+		}
+		tableInfo, err := p.is.TableByName(schema, stmt.ReferTable.Name)
+		if err != nil {
+			p.err = err
+			return
+		}
+		if tableInfo.Meta().TempTableType != model.TempTableNone {
+			p.err = ErrOptOnTemporaryTable.GenWithStackByArgs("create table like")
```
Comment (djshow832) [inline planner/core/preprocess.go:599]: What if `stmt.Table` is also a temporary table?

Comment (Howie59) [reply planner/core/preprocess.go:599]: next pr i will add a check

**Pull request merged**